plugins/blueprint.cpp

```diff
@@ -789,8 +798,52 @@ command_result blueprint(color_ostream &out, vector<string> &parameters)
         end.z = -1;
 
     std::ostringstream err;
-    command_result result = do_transform(start, end, options, err);
-    if (result != CR_OK)
+    if (!do_transform(start, end, options, files, err))
+    {
         out.printerr("%s\n", err.str().c_str());
-    return result;
+        return false;
+    }
+    return true;
 }
+
+// entrypoint when called from Lua. returns the names of the generated files
+static int run(lua_State *L)
+{
+    int argc = lua_gettop(L);
+    vector<string> argv;
+
+    for (int i = 1; i <= argc; ++i)
+    {
+        const char *s = lua_tostring(L, i);
+        if (s == NULL)
+            luaL_error(L, "all parameters must be strings");
+        argv.push_back(s);
+    }
+
+    vector<string> files;
+    if (do_blueprint(Core::getInstance().getConsole(), argv, files))
+    {
+        Lua::PushVector(L, files);
+        return 1;
+    }
+
+    return 0;
+}
+
+command_result blueprint(color_ostream &out, vector<string> &parameters)
+{
+    vector<string> files;
+    if (do_blueprint(Core::getInstance().getConsole(), parameters, files))
```

Comment:
This should technically use the existing output stream, for better behavior when run from other locations (command-prompt, dfhack-run, etc.)
```suggestion
    if (do_blueprint(out, parameters, files))
```

Reply:
Will fix, but calling getConsole is a paradigm that i've seen (and used) in many other places in the code. I'll be sure to use the passed-in `out` object or `Lua::getOutput()` in the future

Reply:
No problem. It may not be necessary in the Lua case, since a GUI script doesn't really have an output stream other than the console anyway, but if you have an `out` parameter, that's generally what you'd want to go with. I know there are several uses of `getConsole()` in other plugins and core code because passing `out` parameters around was either deemed too messy or just didn't make sense, so there are definitely exceptions.

Reply:
done